pandas/tests/indexes/test_numpy_compat.py

```diff
@@ -74,14 +74,16 @@ def test_numpy_ufuncs_basic(index, func):
 @pytest.mark.parametrize(
     "func", [np.isfinite, np.isinf, np.isnan, np.signbit], ids=lambda x: x.__name__
 )
-def test_numpy_ufuncs_other(index, func):
+def test_numpy_ufuncs_other(index, func, request):
     # test ufuncs of numpy, see:
     # https://numpy.org/doc/stable/reference/ufuncs.html
 
     if isinstance(index, (DatetimeIndex, TimedeltaIndex)):
         if isinstance(index, DatetimeIndex) and index.tz is not None:
             if func in [np.isfinite, np.isnan, np.isinf]:
-                pytest.xfail(reason="__array_ufunc__ is not defined")
+                if not np_version_under1p17:
```

Comment:
why do we not prefer an inside use of `pytest.xfail` and this format?

Reply:
using pytest.xfail breaks out immediately, so you dont get alerted if the xfail has been fixed.

Reply:
interesting; it doesn't react to strict=True?

Reply:
AFAIK thats only when used as a decorator `@pytest.mark.xfail(strict=foo)`

Reply:
kk

can u update the way to write tests in the docs (open an issue or PR)

also prob need to have a rule which prohibits (open an issue)